src/crab/array_domain.cpp

```
@@ -206,18 +206,26 @@ std::vector<Cell> offset_map_t::get_overlap_cells(const offset_t o, const unsign
+// the cell set.
+class StackCellRegistry final {
+    std::unordered_map<DataKind, offset_map_t> _maps;
 
-static thread_local LazyAllocator<array_map_t> thread_local_array_map;
+  public:
+    offset_map_t& get(const DataKind kind) { return _maps[kind]; }
+    void clear() { _maps.clear(); }
+};
 
-void clear_thread_local_state() { thread_local_array_map.clear(); }
+void StackCellRegistryDeleter::operator()(StackCellRegistry* p) const noexcept { delete p; }
```

Clang (14.0.6)

[warning] parameter name 'p' is too short, expected at least 3 characters

(readability-identifier-length)